test/integration/017_runtime_materialization_tests/test_runtime_materialization.py

```diff
@@ -56,23 +55,23 @@ def test_postgres_full_refresh(self):
     @use_profile('postgres')
     def test_postgres_delete__dbt_tmp_relation(self):
         # This creates a __dbt_tmp view - make sure it doesn't interfere with the dbt run
-        self.run_sql_file("create_view__dbt_tmp.sql")
+        #self.run_sql_file("create_view__dbt_tmp.sql")
```

Comment:
Ok, I think I get what's going on here:
- During its materializations, dbt creates these temp + backup objects
- This test ensures that dbt first drops any temp/backup objects by the same name, if they already exist
- Previously, those temp + backup objects had predictable names/suffixes (`view__dbt_tmp`, `view__dbt_backup`)
- Thanks to the changes in this PR, those temp + backup objects have nondeterministic names like `view__dbt_tmp095351072872` and `view__dbt_backup095351077674`

That's better! It's much much less likely for these objects to accidentally collide with a preexisting object (almost impossible). So it makes this test _much_ trickier to define, at the same time, it makes the test significantly less important. I think we can delete this test case (`test_postgres_delete__dbt_tmp_relation`) and explain why.

Reply:
That as my thinking precisely when Elize and I talked. Glad you concur :)

Reply:
Triple checking that it's appropriate to have backup objects with nondeterministic names as proposed by this PR. I had originally removed that behavior [in the second commit](https://github.com/dbt-labs/dbt-core/pull/4921/commits/d985a8d7387cc98e8335a5c0950a4060e7fd6691) based on [this](https://github.com/dbt-labs/dbt-core/pull/2850#issuecomment-717672329) comment & the failing test, but after the suffix macro recommendations I added it back. Happy to go in either direction, just let me know what y'all think is best @jtcohen6 @VersusFacit and I'll either remove `test_postgres_delete__dbt_tmp_relation` or differentiate the behavior between backup/temp. Once that decision is made, I think the PR is ready for review 🥳

Reply:
@epapineau Appreciate the triple check! After a bit more thought, I think it might be _very slightly_ preferable to keep relation names non-deterministic _if_ they're creating permanent objects (views + non-temp tables). That way, we avoid cluttering the schema if the materialization fails during the alter-rename-swap-drop step. Deterministically named temp/backup objects get dropped the next time dbt runs that model/etc; actual temp tables get dropped automatically at the end of the session.

That said, failure in that step is _far_ less likely than in the `create table/view as` step, and it's a problem that exists today anyway, whenever users rename a model, and the "old" name of that model doesn't get dropped / cleaned up in the database.

If we did want to keep that distinction—between the temp identifiers used for true temp tables, and our very not-temporary "temporary"/"backup" relations—I think it would just look like one extra argument to these macros that disables the inclusion of `dtstring` in the suffix. That wouldn't be so hard, but it would require a little extra plumbing. It means we could keep this test in place.

What do you think?

Reply:
Alright, done~✨ PR is marked as ready for review.